repose-aggregator/docs/src/asciibinder/services/http-logging.adoc

```diff
@@ -0,0 +1,310 @@
+= HTTP Logging Service
+
+The HTTP Logging Service enables logging of operator-configurable messages for every HTTP interaction.
+
+Usage of this service is just a matter of configuring messages.
+Messages tell this service what to log and where to log it.
+Messages include a template to define what should be written to a log.
+Templates will be populated and rendered using data relevant to each HTTP interaction.
+
+This service can be used for auditing all interactions with an API.
+
+== Configuration
+
+* *Default Configuration:* http-logging.cfg.xml
+* *Released:* TBD
+* link:../schemas/http-logging.xsd[Schema]
+
+=== Full Configuration
+
+This configuration sets every available setting explicitly for the sake of detailing the available settings.
+
+[source,xml]
+.http-logging.cfg.xml
+----
+<http-logging xmlns="http://docs.openrepose.org/repose/http-logging/v1.0">
+    <message <!--1-->
+        log-to="plain-access-log" <!--2-->
+        format="plain"> <!--3-->
+        Request handled!
+    </message>
+</http-logging>
+----
+<1> Declares a message to be logged for every HTTP interaction.
+    The value of this element serves as the content of the message.
+    The value will be processed as a template.
+    See <<Templating>> for more information.
+<2> Specifies the name of the logger to which messages should be logged.
+    See <<../architecture/logging.adoc#, Logging>> for more information.
+<3> Specifies the format of the content.
+    The format determines if and how to validate the content and how to escape values populated by the templating engine.
```

Comment:
I was looking for the `Default` and `Options` for the `format` attribute.
I realize they are in the schema, but this is something that has been being put here as well since converting to the new versioned docs.
The main concern here is that it is another change and we haven't finished fully converting all of the old docs yet.
Maybe something to talk about.

Reply:
Something like these:
- Callout **4** [here](http://www.openrepose.org/versions/9.0.0.0/services/atom-feed-consumption.html#full-configuration)
- Callout **14** & **15**  [here](http://www.openrepose.org/versions/9.0.0.0/filters/api-validator.html#all-the-features-of-a-validator-element)

Reply:
Quoted most of what I wrote on Slack:

> On my first pass I included the default and options, then started adding multiplicity for the `message` element. As I was doing so, I started thinking that listing all of that in the example configuration docs was quite verbose and technically redundant with the schema. So I ended up dropping all of that information with the intent of making examples which would serve to show what can be done with the configuration, and if a user needs full details for some reason, the schema is linked on the page anyway. 

There is also the issue of having multiple sources of truth: one in the schema, and one in the documentation. 

However, I agree that this is inconsistent with existing docs, which does not make for a good user experience. Additionally, if/when we move away from XML for configuration we may lose schemas anyway, in which case we still need a way to give users a complete view of acceptable configuration.

So point taken. I will add the default and valid values where applicable to remain consistent with our current approach, but I am glad to have tried this out and to have discussed  it.